build/cmake/DefineOptions.cmake

```diff
@@ -164,31 +164,36 @@ message(STATUS "Thrift package version:                       ${PACKAGE_VERSION}
 message(STATUS "Build configuration Summary")
 message(STATUS "  Build Thrift compiler:                      ${BUILD_COMPILER}")
 message(STATUS "  Build compiler plugin support:              ${WITH_PLUGIN}")
+if (WITH_PLUGIN_COMPILER_TOO_OLD)
+  MESSAGE_DEP(WITH_PLUGIN "Disabled due to older compiler")
+else()
+  MESSAGE_DEP(WITH_PLUGIN "Disabled by WITH_PLUGIN=OFF")
```

Comment:
I mean, for example, `WITH_CPP` is a flag that users can directly set ON/OFF while `BUILD_CPP` is an internal flag that is based on `WITH_CPP` but can be disabled by `Boost-FOUND=OFF`.

The problem for `WITH_PLUGIN` is that, we're directly modifying this flag without internal equivalent, based on `BUILD_CPP` and `BUILD_COMPILER` values.
So we cannot tell what ON/OFF value the user has put to it.

About autotools flag, try `--enable-plugin=no` instead.
Internally inside configure.ac, with/without flags seemed to be tied to libraries so I used `enable` flag instead.